internal/broker/server.go

```diff
@@ -405,22 +424,132 @@ func (srv *Server) bindAction(w http.ResponseWriter, r *http.Request) {
 		}
 		logRespFields := logrus.Fields{
 			"action":                "bind",
-			"resp:async":            false,
+			"resp:async":            sResp.Async,
 			"resp:credentials:keys": keys,
 		}
 		srv.logger.WithFields(logRespFields).Info("action response")
 	}
 
-	egDTO := BindSuccessResponseDTO{
-		Credentials: sResp.Credentials,
+	if !sResp.Async {
+		//logResp(logRespFields)
+		egDTO := BindSuccessResponseDTO{
+			Credentials: sResp.Credentials,
+		}
+		srv.writeResponse(w, http.StatusOK, egDTO)
+		return
 	}
-	srv.writeResponse(w, http.StatusCreated, egDTO)
+
+	opID := internal.OperationID(*sResp.OperationKey)
+	egDTO := BindInProgressResponseDTO{Operation: &opID}
+
+	srv.writeResponse(w, http.StatusAccepted, egDTO)
 }
 
 func (srv *Server) unBindAction(w http.ResponseWriter, r *http.Request) {
 	srv.writeResponse(w, http.StatusGone, map[string]interface{}{})
 }
 
+func (srv *Server) getServiceBinding(w http.ResponseWriter, r *http.Request) {
+	osbCtx, _ := osbContextFromContext(r.Context())
+
+	instanceID := mux.Vars(r)["instance_id"]
+	bindingID := mux.Vars(r)["binding_id"]
+
+	sReq := osb.GetBindingRequest{
+		InstanceID: instanceID,
+		BindingID:  bindingID,
+	}
+
+	sResp, err := srv.binder.GetServiceBinding(r.Context(), osbCtx, &sReq)
```

Comment:
You have define a binder interface with method GetServiceBinding which is returning HTTPStatusCodeError.It is not possible the function used below (in switch statement) `IsNotFoundError` is returning true.
The pattern used here is the service method returns an error, which is returning the error from one of the storage package. Such error could be tested by the IsNotFoundError.